### PR TITLE
Alter transaction checks for starknet-rs/foundry compatibility

### PR DESCRIPTION
**Pull request by justinfrevert**
This PR fixes errors which occur when sending transactions through Starknet Foundry/starknet.rs.

## What is the current behavior?

Currently, transactions sent from starknet-rs fail, since starknet-rs sends `is_query: false` for their estimate calls.  

Resolves: #1168 

## What is the new behavior?

- Remove the `is_query` check in the estimate function. According to the chat in the starknet-rs, this will be changed later(see [the discussion](https://t.me/starknet_rs/4880)) so I added an error log indicating this. 
- Consequently, `is_query` can't be defaulted further along the code path since it is used in the construction of the hash used in signing/verifying. We accept the `is_query` value that was sent instead.

## Does this introduce a breaking change?

No, because the `is_query` value which was sent is used.

## Other information

This is also related to https://github.com/keep-starknet-strange/madara/issues/1149. That work will be done in a separate pr

### CODE REVIEW

**Comment by justinfrevert** [comment]
It seems like some of the actions(declare) still fail at the extrinsic-level. I was a bit confused because of the tool's output. I need to keep looking into it

**Comment by tdelabro** [comment]
@justinfrevert is it ready for review?

File: crates/client/rpc/src/lib.rs [hunk 1]
```diff
@@ -608,7 +608,9 @@ where
             BroadcastedTransaction::DeployAccount(deploy_tx) => !deploy_tx.is_query,
         });
         if is_invalid_query_transaction {
```
Comment (tdelabro) [inline crates/client/rpc/src/lib.rs:610]: reverse the bool:
- remove all the `!` in the match
- rename `is_invalid_query_transaction` to `is_query`
- change `if is_invalid_query_transaction` to `if !is_query {}`
-  call `.estimate_fee(substrate_block_hash, tx, is_query)`

It's better than taking the inverse of the inverse

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: crates/client/rpc/src/lib.rs (hunk 1 above)
Comment (justinfrevert) [reply crates/client/rpc/src/lib.rs]: That's nicer, thanks!

**Comment by justinfrevert** [comment]
> @justinfrevert is it ready for review?

Yes, it was at the time. For some reason I wasn't able to put a label on it when I created it.

**Comment by justinfrevert** [comment]
Also, fyi: I have a github action in the works to assure this functionality. I am wrapping that work up and will submit a separate PR soon.

**Review by tdelabro — APPROVED** [review]

**Pull request merged**